src/loader.js

```diff
@@ -10,9 +11,13 @@ const reactModule = /['"]react['"]/;
  * @returns {string} The injected module source code.
  */
 function RefreshHotLoader(source) {
+  // Add dependency to allow caching and invalidations
+  this.addDependency(path.resolve('./runtime/RefreshModuleRuntime'));
+
   // Only apply transform if the source code contains a React import
   return reactModule.test(source)
     ? source +
```

Comment:
I think this breaks source maps. You probably want to use the "source" abstraction instead which preserves mappings. See how webpack official loaders do it.

Reply:
I don't think the fix is difficult btw. I think you just need to use ConcatSource or something like this instead of actual concatenation.

Reply:
Can you test again with my latest patch? It should fix the issues with source maps. The issue was not properly accepting and passing source maps through.

Reply:
I had thought about using Dependencies, Sources and Templates like [this answer on StackOverflow](https://stackoverflow.com/a/52906440), but injecting external dependencies (`react-refresh`) this way was almost impossible without a full re-parse, so I went with a loader instead.